### PR TITLE
WPF - Fix Width and Height measurement when GetDesiredSize

### LINKED ISSUE #1588: wpf stacklayout  WidthRequest adds unwanted margin
### Description
when setting stacklayout  WidthRequest it adds unwanted margin 

### Expected Behavior
stacklayout without the extra margin

### Actual Behavior
stacklayout with extra margin
### Basic Information

- Version with issue: xamarin.forms 2.6.0.91472-nightly
- IDE: vs2017
- Platform: WPF
- Affected Devices: Windows 10

### Screenshots

wpf

### PR DESCRIPTION
**Pull request by mohachouch**
### Description of Change ###
Fix Width and Height measurement when GetDesiredSize

### Issues Resolved ### 
- fixes #1588, #3308, #3558 

### Platforms Affected ### 
- WPF

### Screen ###

image

### CODE REVIEW

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3558.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 3558, "[WPF] Label displays only one line", PlatformAffected.WPF)]
+	public class Issue3558 : TestContentPage
+	{
+		protected override void Init()
+		{
+			var stack = new StackLayout();
+			Label label = new Label { Text = "Lotsa test here to make this a really really really really long line that should be wrapped",
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3558.cs]: "test" -> "text"

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+		{
+			set { SetValue(ThicknessProperty, value); }
+			get { return (double)GetValue(ThicknessProperty); }
+		}
+
+		public Divider()
+		{
+			line = new BoxView();
+			Content = line;
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+				divider.Color = (Color)newvalue;
+			});
+
+		public Color Color
+		{
+			set { SetValue(ColorProperty, value); }
+			get { return (Color)GetValue(ColorProperty); }
+		}
+
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+		}
+	}
+
+	public enum DividerOrientation
+	{
+		Vertical,
+
+		Horizontal
+	}
+	class Divider : ContentView
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Missing blank line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+
+		protected override void Init()
+		{
+		}
+	}
+
+	public enum EntryOrientation
+	{
+		Vertical,
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+						<local:LabledEntry LabelText="Telephone" LabelTextColor="#757575" />
+						<local:LabledEntry LabelText="Email" LabelTextColor="#757575" />
+						<local:LabledEntry LabelText="Website" LabelTextColor="#757575" />
+					</StackLayout>
+
+					<StackLayout WidthRequest="300">
+						<Label Text="TAX INFORMATION" FontAttributes="Bold" TextColor="#212121" Margin="0,0,0,20"/>
+						<local:LabledEntry LabelText="Tax card" LabelTextColor="#757575" />
+						<local:LabledEntry LabelText="Tax file no." LabelTextColor="#757575" />
+						<local:LabledEntry LabelText="Commerial regestration" LabelTextColor="#757575" />
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml]: "Commerial regestration" -> "Commercial Registration"

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+		Horizontal
+	}
+
+	class LabledEntry : ContentView
+	{
+		StackLayout stk;
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Private members should use `_` prefix.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+				labledEntry.Orientation = orientation;
+			});
+
+		public EntryOrientation Orientation
+		{
+			set { SetValue(OrientationProperty, value); }
+			get { return (EntryOrientation)GetValue(OrientationProperty); }
+		}
+
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra blank line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+				labledEntry.Placeholder = (string)newvalue;
+			});
+
+		public string Placeholder
+		{
+			set { SetValue(PlaceholderProperty, value); }
+			get { return (string)GetValue(PlaceholderProperty); }
+		}
+
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra blank line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+				labledEntry.EntryTextColor = (Color)newvalue;
+			});
+
+		public Color EntryTextColor
+		{
+			set { SetValue(EntryTextColorProperty, value); }
+			get { return (Color)GetValue(EntryTextColorProperty); }
+		}
+
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra blank line

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,349 @@
...
+			entry = new Entry();
+			entry.FontSize = Device.GetNamedSize(NamedSize.Small, entry);
+			label = new Label();
+			entry.FontSize = Device.GetNamedSize(NamedSize.Small, label);
+			entry.HorizontalOptions = LayoutOptions.FillAndExpand;
+			label.VerticalOptions = LayoutOptions.Center;
+			stk.Children.Add(label);
+			stk.Children.Add(entry);
+			Content = stk;
+
```
Comment (hartez) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1588.xaml.cs]: Extra blank line

**Comment by mohachouch** [comment]
Done

**Comment by rmarinho** [comment]
build

**Comment by rmarinho** [comment]
Xaml samples on the gallery need some directives to run when build as shared on UITest, that's why build is failing , check this example:

https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla23942.xaml.cs#L33

**Comment by samhouts** [comment]
build

**Pull request merged**